### DISCUSSION TITLE
How to dynamically add chapter thumbnails?

### DISCUSSION THREAD

**Discussion by INaWolf** [vidstack/player#1033] [Q&A] [answered]
Hi everyone, I have a question about the chapters and the thumbnails.
I already figured out how to dynamically add chapters with 'titles' on the video without having a .vtt file (i am using VueJS):
```ts
const playerRef = ref<MediaPlayer>();
onMounted(() => {

  const track = new TextTrack({
    kind: 'chapters',
    type: 'vtt',
    default: true
  });
  track.mode = 'showing';

  track.addCue(new VTTCue(0, 50, 'My first chapter'));
  track.addCue(new VTTCue(51, 500, 'My second chapter'));
  playerRef.value?.textTracks.add(track);

});
```

With my player defined as:
```html
<media-player ref="playerRef" :title="video?.name" :src="mysource">
  <media-provider>
  </media-provider>
</media-player>
```

However, I cannot figure out how to add chapter image to each chapter text?
It might not be possible, since I've seen that it will automatically map the nearest thumbnails onto the chapters, but if I could just map the generated image at the specified timecode for each chapter it would be cool.

Thanks for your help!

**Comment by mihar-22** [comment, marked as answer]
Hey there! There's no way at the moment to specify a thumbnail for a specific chapter. As you've noticed we map the thumbnails to the chapters automatically based on start time. This can be a feature request which you can open a new issue for if you like :)

**Reply by INaWolf** [reply]
Ok no problem, thank you for your answer!

I'll fill a pull request when I'll have time.

Thank you again :)
